src/wazuh_modules/agent_upgrade/wm_agent_parsing.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Wazuh Module for Agent Upgrading
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * July 3, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "wazuh_modules/wmodules.h"
+#include "wm_agent_parsing.h"
+#include "wm_agent_upgrade_tasks.h"
+
+int wm_agent_parse_command(const char* buffer, cJSON** json_api, cJSON** params, cJSON** agents) {
+    int retval = -1;
+    cJSON * root = cJSON_Parse(buffer);
+    if (!root) {
+        mterror(WM_AGENT_UPGRADE_LOGTAG, WM_UPGRADE_JSON_PARSE_ERROR,  buffer);
+        *json_api = wm_agent_parse_response_mesage(PARSING_ERROR, upgrade_error_codes[PARSING_ERROR], NULL, NULL, NULL);
+    } else {
+        cJSON *command = cJSON_GetObjectItem(root, "command");
+        *params = cJSON_GetObjectItem(root, "params");
+        *agents = cJSON_GetObjectItem(root, "agents");
+        if (command && *agents) {
+            *json_api = root;
+            if (strcmp(command->valuestring, WM_AGENT_UPGRADE_COMMAND_NAME) == 0) {
+                retval = 0;
+            } else if (strcmp(command->valuestring, WM_AGENT_UPGRADE_RESULT_COMMAND_NAME) == 0) { 
+                retval = 1;
+            } else {
+                // TODO invalid command
+                mterror(WM_AGENT_UPGRADE_LOGTAG, WM_UPGRADE_UNDEFINED_ACTION_ERRROR,  command->valuestring);
+                json_api = wm_agent_parse_response_mesage(TASK_CONFIGURATIONS, upgrade_error_codes[TASK_CONFIGURATIONS], NULL, NULL, NULL);
+            }
+        } else {
+            mterror(WM_AGENT_UPGRADE_LOGTAG, WM_UPGRADE_REQUIRED_PARAMETERS);
+            *json_api = wm_agent_parse_response_mesage(PARSING_REQUIRED_PARAMETER, upgrade_error_codes[PARSING_REQUIRED_PARAMETER], NULL, NULL, NULL);
+            cJSON_Delete(root);
+        }
+    }
+    return retval;
+}
+
+wm_upgrade_task* wm_agent_parse_upgrade_command(const cJSON* params, char* output) {
+    wm_upgrade_task *task = wm_agent_init_upgrade_task();
+    int param_index = 0;
+    int error_flag = 0;
+    while(!error_flag && (param_index < cJSON_GetArraySize(params))) {
```

Comment:
Check if params is not NULL.

Reply:
If `params` is NULL, is the upgrade valid? Or should it be stopped?

Reply:
It is a valid option, it will use everything by default. Wazuh repository and latest version, so we don't want it to fail in that case